src/modules/Dashboard/features/RespondentData/RespondentData.utils.ts

```diff
@@ -11,3 +14,25 @@ export const createArrayForSlider = ({
     value: minValue + index,
     label: minValue + index,
   }));
+
+export const getUniqueIdentifierOptions = (identifiers: Identifier[]) =>
+  identifiers.reduce((uniqueIdentifiers: AutocompleteOption[], identifierItem) => {
+    if (!identifierItem) return uniqueIdentifiers;
+
+    const { decryptedValue } = identifierItem;
+
+    if (
+      uniqueIdentifiers &&
+      !uniqueIdentifiers.find((identifier) => identifier.id === decryptedValue)
```

Comment:
Please use the `Set` instance instead to not iterate the array due to `find` work.